engine/src/main/java/org/hibernate/search/v6poc/backend/document/model/ObjectFieldStorage.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.v6poc.backend.document.model;
+
+public enum ObjectFieldStorage {
+
+	/**
+	 * Use the backend-specific default.
+	 */
+	DEFAULT,
```

Comment:
Do we really expect to have different behavior per backend? I mean it would really change the results so I'm not sure it's such a good idea (we can revisit that later though).

Reply:
It doesn't change the results, it only changes whether users can use "nested" queries or not. Without a "nested" query, the "nested" field behaves exactly as an "object" field.
FWIW, the main use I saw for this was the ability for users to specify a default for storage in multi-valued `@IndexedEmbedded`: some people may always want to use "NESTED" storage.

Reply:
> It doesn't change the results

Well, changing the storage format does change the search results (see your example with John Doe / Jane Smith).

But the main use you have in mind convinces me it's a good idea: it's a nice thing to be able to override the default at a global level.